var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1132,82 +870,306 @@ def setup_dependent_build_environment(self, env, dependent_spec):
 
-    def remove_files_from_view(self, view, merge_map):
-        bin_dir = self.spec.prefix.bin if sys.platform != "win32" else self.spec.prefix
-        for src, dst in merge_map.items():
-            if not path_contains_subdirectory(src, bin_dir):
-                view.remove_file(src, dst)
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def import_tests(self):
+        """Test that basic Python functionality works."""
+
+        spec = self.spec
+
+        with working_dir("spack-test", create=True):
+            # Ensure that readline module works
+            if "+readline" in spec:
+                self.pkg.command("-c", "import readline")
+
+            # Ensure that ssl module works
+            if "+ssl" in spec:
+                self.pkg.command("-c", "import ssl")
+                self.pkg.command("-c", "import hashlib")
+
+            # Ensure that sqlite3 module works
+            if "+sqlite3" in spec:
+                self.pkg.command("-c", "import sqlite3")
+
+            # Ensure that dbm module works
+            if "+dbm" in spec:
+                self.pkg.command("-c", "import dbm")
+
+            # Ensure that nis module works
+            if "+nis" in spec:
+                self.pkg.command("-c", "import nis")
+
+            # Ensure that zlib module works
+            if "+zlib" in spec:
+                self.pkg.command("-c", "import zlib")
+
+            # Ensure that bz2 module works
+            if "+bz2" in spec:
+                self.pkg.command("-c", "import bz2")
+
+            # Ensure that lzma module works
+            if "+lzma" in spec:
+                self.pkg.command("-c", "import lzma")
+
+            # Ensure that pyexpat module works
+            if "+pyexpat" in spec:
+                self.pkg.command("-c", "import xml.parsers.expat")
+                self.pkg.command("-c", "import xml.etree.ElementTree")
+
+            # Ensure that ctypes module works
+            if "+ctypes" in spec:
+                self.pkg.command("-c", "import ctypes")
+
+            # Ensure that tkinter module works
+            # https://wiki.python.org/moin/TkInter
+            if "+tkinter" in spec:
+                # Only works if ForwardX11Trusted is enabled, i.e. `ssh -Y`
+                if "DISPLAY" in env:
+                    self.pkg.command("-c", "import tkinter; tkinter._test()")
+                else:
+                    self.pkg.command("-c", "import tkinter")
+
+            # Ensure that uuid module works
+            if "+uuid" in spec:
+                self.pkg.command("-c", "import uuid")
+
+            # Ensure that tix module works
+            if "+tix" in spec:
+                self.pkg.command("-c", "import tkinter.tix")
+
+            # Ensure that crypt module works
+            if "+crypt" in spec:
+                self.pkg.command("-c", "import crypt")
+
+
+class AutotoolsBuilder(autotools.AutotoolsBuilder, RunAfter, BuildEnvironment):
```

Comment:
According to #36825, the builder base class should actually come last (@alalazo what's the reason for this, and is it possible to make it order invariant?)